Feat(UI) #9204: Added preview panel for all the data assets.

aniketkatkar97 opened the pull request:
### Describe your changes :
<!-- Explain what you have done & tag your assigned issue !-->
I worked on #9204 

- [x] Adding data assets summary details panel for all the data assets.
- [x] Styling improvements
- [x] bug fixes

#
### Type of change :
<!-- You should choose 1 option and delete options that aren't relevant -->
- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
<p align="center">

1. Tables:
<img width="1154" alt="Screenshot 2022-12-16 at 11 51 30 AM" src="https://user-images.githubusercontent.com/51777795/208035560-6d0075c8-2295-482a-8456-17b4a4ba0500.png">

2. Dashboards:
<img width="1154" alt="Screenshot 2022-12-16 at 11 51 39 AM" src="https://user-images.githubusercontent.com/51777795/208035579-ef0449a8-3e96-4b59-bcdc-a2d5758daa54.png">

3. Pipelines:
<img width="1156" alt="Screenshot 2022-12-16 at 11 51 45 AM" src="https://user-images.githubusercontent.com/51777795/208035624-23b28e66-4862-4c05-ba2d-15dc24eea279.png">

4. MlModels:
<img width="1156" alt="Screenshot 2022-12-16 at 11 51 54 AM" src="https://user-images.githubusercontent.com/51777795/208035638-ece72a3e-abbe-42c6-b6aa-884dfe35dd6e.png">

</p>

#
### Checklist:
<!-- add an x in [] if done, don't mark items that you didn't do !-->
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
<!-- Please see the contributing guidelines and then add your reviewer(s) !-->
<!--- OpenMetadata community thanks you for explaining your changes in detail !-->
<!--- If you are unsure of people to review your work, you can add anyone of these developers :) !-->
Frontend: @open-metadata/ui 
<!--- Backend: @open-metadata/backend -->
<!--- Ingestion: @open-metadata/ingestion -->

chirag-madlani commented:
@aniketkatkar97 Did we also make the first result selected by default?

chirag-madlani commented on openmetadata-ui/src/main/resources/ui/src/components/Explore/EntitySummaryPanel/EntitySummaryPanel.component.tsx:
```diff
@@ -12,109 +12,59 @@
+    if (entityDetails.entityType === ExplorePageTabs.TABLES) {
+      return <TableSummary entityDetails={entityDetails.details as Table} />;
+    } else if (entityDetails.entityType === ExplorePageTabs.TOPICS) {
+      return <TopicSummary entityDetails={entityDetails.details as Topic} />;
+    } else if (entityDetails.entityType === ExplorePageTabs.DASHBOARDS) {
+      return (
+        <DashboardSummary entityDetails={entityDetails.details as Dashboard} />
+      );
+    } else if (entityDetails.entityType === ExplorePageTabs.PIPELINES) {
+      return (
+        <PipelineSummary entityDetails={entityDetails.details as Pipeline} />
```
use switch case for such case

chirag-madlani commented on openmetadata-ui/src/main/resources/ui/src/components/Explore/EntitySummaryPanel/SummaryList/SummaryList.interface.ts:
```diff
@@ -0,0 +1,38 @@
+import { ReactNode } from 'react';
```
Should be after licence

chirag-madlani commented on openmetadata-ui/src/main/resources/ui/src/components/Explore/EntitySummaryPanel/TableSummary/TableSummary.component.tsx:
```diff
@@ -0,0 +1,238 @@
+  const overallSummary: OverallTableSummeryType[] = useMemo(() => {
+    return [
+      {
+        title: 'Row Count',
```
Missed localisation here?

aniketkatkar97 commented:
> @aniketkatkar97 Did we also make the first result selected by default?
Yes we did, also the selected card will remain highlighted

aniketkatkar97 commented:
> @devyani-kaushik the side bar headings compare to the table name heading is bigger. May be we should make it smaller to get attention to the table name?

@harshach, Changed the styling for the summary title also, updated the screenshots. Can you please check?
